For FreeBSD, libxslt builds libraries as libxslt.so.2 or something
like that. Following patch fixes this.

Index: libxslt/libxslt-1.1.16.patch
===================================================================
RCS file: /cvs/external/libxslt/libxslt-1.1.16.patch,v
retrieving revision 1.2
diff -u -r1.2 libxslt-1.1.16.patch
--- libxslt/libxslt-1.1.16.patch	10 May 2007 13:10:07 -0000	1.2
+++ libxslt/libxslt-1.1.16.patch	20 May 2007 02:35:31 -0000
@@ -108,3 +108,46 @@
 ! 
 ! exit 0
 
+*** misc/libxslt-1.1.16/ltmain.sh	Sun Feb 12 09:50:49 2006
+--- misc/build/libxslt-1.1.16/ltmain.sh	Sun May 20 11:34:29 2007
+***************
+*** 3192,3200 ****
+  	    revision="$number_revision"
+  	    ;;
+  	  freebsd-aout|freebsd-elf|sunos)
+! 	    current="$number_major"
+! 	    revision="$number_minor"
+! 	    age="0"
+  	    ;;
+  	  irix|nonstopux)
+  	    current=`expr $number_major + $number_minor - 1`
+--- 3192,3200 ----
+  	    revision="$number_revision"
+  	    ;;
+  	  freebsd-aout|freebsd-elf|sunos)
+! 	    current=`expr $number_major + $number_minor`
+! 	    age="$number_minor"
+! 	    revision="$number_revision"
+  	    ;;
+  	  irix|nonstopux)
+  	    current=`expr $number_major + $number_minor - 1`
+***************
+*** 3267,3274 ****
+  	  ;;
+  
+  	freebsd-elf)
+! 	  major=".$current"
+! 	  versuffix=".$current";
+  	  ;;
+  
+  	irix | nonstopux)
+--- 3267,3274 ----
+  	  ;;
+  
+  	freebsd-elf)
+! 	  major=.`expr $current - $age`
+! 	  versuffix="$major.$age.$revision"
+  	  ;;
+  
+  	irix | nonstopux)
+
